platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DefaultResolvedArtifact.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.artifacts;
+
+import org.gradle.api.artifacts.ModuleVersionIdentifier;
+import org.gradle.api.artifacts.ResolvedArtifact;
+import org.gradle.api.artifacts.ResolvedModuleVersion;
+import org.gradle.api.artifacts.component.ComponentArtifactIdentifier;
+import org.gradle.api.internal.artifacts.ivyservice.modulecache.dynamicversions.DefaultResolvedModuleVersion;
+import org.gradle.internal.component.model.IvyArtifactName;
+import org.gradle.internal.model.CalculatedValue;
+
+import javax.annotation.Nullable;
+import java.io.File;
+
+/**
+ * Default implementation of {@link ResolvedArtifact}, the artifact type used by the legacy
+ * {@link org.gradle.api.artifacts.ResolvedConfiguration} API. This class presents a file extension, type,
+ * classifier via its {@link IvyArtifactName}. This name is tracked on a best-effort basis, and may not
```

Comment:
🤔 "may not always reqpresent the actual file name" sounds concerning.  Is it worth writing up an example here?  Or, better yet, a test demonstrating this?

Reply:
Consider this build script:
```
            configurations {
                consumable("con") {
                    outgoing.artifact(file("foo.txt")) {
                        classifier = "HELLO"
                    }
                    attributes {
                        attribute(Category.CATEGORY_ATTRIBUTE, objects.named(Category, Category.LIBRARY))
                    }
                }
                dependencyScope("implementation")
                resolvable("res") {
                    extendsFrom(implementation)
                    attributes {
                        attribute(Category.CATEGORY_ATTRIBUTE, objects.named(Category, Category.LIBRARY))
                    }
                }
            }

            dependencies {
                implementation project(":")
            }

            task resolve {
                doLast {
                    def artifact = configurations.res.resolvedConfiguration.resolvedArtifacts.first()
                    println artifact.file.name
                    println artifact.classifier
                }
            }
```

The output is:
```
foo.txt
HELLO
``

That is clearly wrong

Reply:
By "here" I meant in the javadoc itself.  Or adding exactly this as a test would be perfect.

Reply:
I can add this test